public/javascripts/needleedit.js

```diff
@@ -0,0 +1,119 @@
+/*
+ * This file contains all the functions for steps/edit - together with needleeditor.js
+ */
+
+function loadBackground() {
+    nEditor.LoadBackground($(this).data('url'));
+    $("#needleeditor_image").attr("value", $(this).data('image'));
+    $("#needleeditor_imagedistri").attr("value", $(this).data('distri'));
+    $("#needleeditor_imageversion").attr("value", $(this).data('version'));
+}
+
+function loadTagsAndName() {
+    var tags = $(this).data('tags');
+    $("#needleeditor_tags").find('input').each(function() {
+	$(this).attr('checked', tags.indexOf($(this).val()) != -1);
+    });
+    $("#needleeditor_name").val($(this).data('suggested'));
+}
+
+function addTag() {
+    var input = $('#newtag');
+    var checkbox = nEditor.AddTag(input.val(), false);
+    input.val('');
+    checkbox.click();
+    return false;
+}
+
+function do_overwrite()
+{
+    saveNeedleForm = document.forms['save_needle_form'];
+    saveNeedleForm.submit();
+    return true;
+}
+
+function setMargin() {
+    console.log("SETMAR");
+    console.log($('#margin'));
+    console.log($('#margin').val());
+    nEditor.setMargin($('#margin').val());
+}
+
+function setMatch() {
+    nEditor.setMatch($('#match').val());
+}
+
+var nEditor;
+
+function setup_needle_editor(imageurl, default_needle)
+{
+    nEditor = new NeedleEditor(imageurl, default_needle);
+
+    $('.tag_checkbox').click(function() {
+	nEditor.changeTag(this.value, this.checked);
+    });
+
+    $('#tag_add_button').click(addTag);
+    $('#newtag').keypress(function() {
+	if (event.keyCode==13)
+	    return addTag();
+    });
+        
+    $('#property_workaround').click(function() { nEditor.changeProperty(this.name, this.checked) });
+    $('.area_selector').click(function() {
+	nEditor.LoadAreas($(this).data('areas'));
+    });
+
+    $('.background_selector').click(loadBackground);
+    $('.tags_selector').change(loadTagsAndName);
+
+    var matchdialog = $( "#change-match-form" ).dialog({
```

Comment:
I tried that, but it's much harder than it sounds (to me) as all the jquery callbacks take global functions accessing nEditor. So I stopped that experiment